operators/src/plot/temporal_vector_line_plot.rs

```diff
@@ -0,0 +1,360 @@
+}
+
+#[async_trait]
+impl<G> PlotQueryProcessor for FeatureAttributeValuesOverTimeQueryProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+{
+    type OutputFormat = PlotData;
+
+    fn plot_type(&self) -> &'static str {
+        FEATURE_ATTRIBUTE_OVER_TIME_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let values = FeatureAttributeValues::<MAX_FEATURES>::default();
+
+        let values = self
+            .features
+            .vector_query(query, ctx)?
+            .fold(Ok(values), |acc, features| async {
+                match (acc, features) {
+                    (Ok(mut acc), Ok(features)) => {
+                        let ids = features.data(&self.params.id_column)?;
+                        let values = features.data(&self.params.value_column)?;
+
+                        for i in 0..features.len() {
+                            let id: String = match ids.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v.to_string(),
+                                FeatureDataValue::NullableInt(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Float(v) => v.to_string(),
+                                FeatureDataValue::NullableFloat(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Text(v) => v.clone(),
+                                FeatureDataValue::NullableText(v) => {
+                                    v.ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                                FeatureDataValue::Category(v) => v.to_string(),
+                                FeatureDataValue::NullableCategory(v) => {
+                                    v.map(|v| v.to_string())
+                                        .ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                            };
+
+                            let value: f64 = match values.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v as f64,
+                                FeatureDataValue::NullableInt(v) => v.map_or(0.0, |v| v as f64), // TODO: NAN better default?
+                                FeatureDataValue::Float(v) => v,
+                                FeatureDataValue::NullableFloat(v) => v.unwrap_or(0.0),
+                                FeatureDataValue::Category(v) => f64::from(v),
+                                FeatureDataValue::NullableCategory(v) => v.map_or(0.0, f64::from),
+                                _ => return Err(Error::InvalidFeatureDataType),
```

Comment:
What about a method `as_f64()` or `as_float()`?

I guess we will need it elsewhere and it improves readability.

Reply:
tbh I much rather would like a way of iterating over a column as a certain type. So turning a FeatureDataRef into an Iterator that automatically converts int to float or int to string etc. I created a card https://trello.com/c/MSNIEFxd

Reply:
Could you extract a function then to improve the readability of the function and then put a link from the trello card to this code block such that we know what to replace once we created the functionality?

Reply:
92801e562d20b900275516583b8b8ab4b228334b